src/randoop/main/GenTests.java

```diff
@@ -183,17 +189,23 @@ public boolean handle(String[] args) throws RandoopTextuiException {
       System.out.println("Use the --classlist, --testclass, or --methodlist options.");
       System.exit(1);
     }
-    
-    
+
+    VisibilityPredicate visibility;
+    if (GenInputsAbstract.public_only) {
+      visibility = new PublicVisibilityPredicate();
+    } else {
+      visibility = new NotPrivateVisibilityPredicate();
+    }
+
     List<Class<?>> allClasses = findClassesFromArgs(options);
 
     // Remove private (non-.isVisible) classes and abstract classes
     // and interfaces.
     List<Class<?>> classes = new ArrayList<>(allClasses.size());
     for (Class<?> c : allClasses) {
-      if (Reflection.isAbstract (c)) {
+      if (Modifier.isAbstract (c.getModifiers()) && !c.isEnum()) {
         System.out.println("Ignoring abstract " + c + " specified via --classlist or --testclass.");
-      } else if (! Reflection.isVisible (c)) {
+      } else if ( !visibility.isVisible (c) ) {
```

Comment:
I would leave the space between the `!` operator and the expression it applies to.  Otherwise it's too easy to overlook, is harder to read (in my opinion), and is inconsistent with elsewhere in the file.
I would remove the added space at the end of the line, however -- again, for consistency.